### PR TITLE
Remove passing of fence state into AFS

### PR DESCRIPTION
**Pull request by peterbarker**
We now have a singleton which all Plane, Rover and Copter all use - so use that singleton in AFS.

I tidied the AFS test and checked it under Copter (needs a few not-included hacks as AFS isn't enabled on Copter by default).

Also removes code in lieu of calling a method which does exactly the same thing.

Note the equivalent behaviour to the old code - if you don't have fence compiled in then you won't breach the geofence, even if your AFS parameters say "check the geofence".  Note the same parameter gates checking the qnh altitude limit, however.

### CODE REVIEW

File: libraries/AP_AdvancedFailsafe/AP_AdvancedFailsafe.cpp [hunk 1]
```diff
@@ -165,21 +165,24 @@ const AP_Param::GroupInfo AP_AdvancedFailsafe::var_info[] = {
...
     if (!_enable) {
         return;
     }
 
+#if AP_FENCE_ENABLED
     // we always check for fence breach
     if(_enable_geofence_fs) {
-        if (geofence_breached || check_altlimit()) {
+        const AC_Fence *ap_fence = AP::fence();
+        if ((ap_fence && ap_fence->get_breaches() != 0) || check_altlimit()) {
```
Comment (andyp1per) [inline libraries/AP_AdvancedFailsafe/AP_AdvancedFailsafe.cpp:178]: I think the style we prefer is ap_fence != nullptr

**Review by andyp1per — APPROVED** [review]

File: libraries/AP_AdvancedFailsafe/AP_AdvancedFailsafe.cpp (hunk 1 above)
Comment (peterbarker) [reply libraries/AP_AdvancedFailsafe/AP_AdvancedFailsafe.cpp]: Yeah, I'd noticed that and was just hoping to slide it through ;-)

Changed it.

**Comment by peterbarker** [comment]
MatekF405:
```
Binary Name      Text [B]        Data [B]     BSS (B)        Total Flash Change [B] (%)      Flash Free After PR (B)
---------------  --------------  -----------  -------------  ----------------------------  -------------------------
antennatracker   0 (0.0000%)     0 (0.0000%)  0 (0.0000%)    0 (0.0000%)                                      340712
ardusub          0 (0.0000%)     0 (0.0000%)  0 (0.0000%)    0 (0.0000%)                                      176228
arducopter-heli  0 (0.0000%)     0 (0.0000%)  0 (0.0000%)    0 (0.0000%)                                       46168
blimp            0 (0.0000%)     0 (0.0000%)  0 (0.0000%)    0 (0.0000%)                                      320996
ardurover        0 (0.0000%)     0 (0.0000%)  0 (0.0000%)    0 (0.0000%)                                      140952
arduplane        -68 (-0.0071%)  0 (0.0000%)  -4 (-0.0031%)  -68 (-0.0071%)                                    19680
arducopter       0 (0.0000%)     0 (0.0000%)  0 (0.0000%)    0 (0.0000%)                                       55720
```

**Review by rmackay9 — COMMENTED** [review]
I'm happy with the Copter changes.

**Pull request merged**